pkg/deployment/inventory.go

```diff
@@ -92,34 +92,19 @@ func GenerateNodeSetInventory(ctx context.Context, helper *helper.Helper,
 					//host.Vars[string.Join(entry, "_gateway_ip")] = res.Gateway
 					host.Vars[entry+"_host_routes"] = res.Routes
 					dnsSearchDomains = append(dnsSearchDomains, res.DNSDomain)
+					host.Vars["dns_search_domains"] = dnsSearchDomains
 				}
-
-				host.Vars["ansible_user"] = getAnsibleUser(instance, nodeName)
-				host.Vars["ansible_port"] = getAnsiblePort(instance, nodeName)
-				host.Vars["management_network"] = getAnsibleManagementNetwork(instance, nodeName)
-				host.Vars["networks"] = getAnsibleNetworks(instance, nodeName)
-
-				ansibleVarsData, err := getAnsibleVars(helper, instance, nodeName)
-				if err != nil {
-					return "", err
-				}
-				for key, value := range ansibleVarsData {
-					host.Vars[key] = value
-				}
-				host.Vars["dns_search_domains"] = dnsSearchDomains
 			}
 		}
+		host.Vars["ansible_user"] = getAnsibleUser(instance, nodeName)
+		host.Vars["ansible_port"] = getAnsiblePort(instance, nodeName)
+		host.Vars["management_network"] = getAnsibleManagementNetwork(instance, nodeName)
+		host.Vars["networks"] = getAnsibleNetworks(instance, nodeName)
 
 		err = resolveNodeAnsibleVars(&node, &host, &ansible.Group{})
 		if err != nil {
 			return "", err
 		}
-
-		ipSet, ok = allIPSets[nodeName]
-		if ok {
-			populateInventoryFromIPAM(&ipSet, host, dnsAddresses)
-		}
```

Comment:
not sure if we should drop this, wdyt @rabi ?

Reply:
No this should be the last and can't be removed. Seems this file is really messed up with the call to populateInventoryFromIPAM() twice and the lots of code duplicated inline.  This needs cleanup.

Reply:
Ok, I left the last one, removed the first one and all of the duplicated logic that was in this function.

Reply:
Proposed https://github.com/openstack-k8s-operators/dataplane-operator/pull/437 to cleanup and simplify the whole thing.